Sources/_StringProcessing/RegexDSL/Match.swift

```diff
@@ -29,48 +29,54 @@ extension RegexProtocol {
       input.base, in: input.startIndex..<input.endIndex)
   }
 
-  // FIXME: This is mostly hacky because we go down two different paths based on
-  // whether there are captures. This will be cleaned up once we deprecate the
-  // legacy virtual machines.
-  func _match(
+  // TODO: Should we expose parameters for testing?
+  // Currently, tests just use the execution interface directly.
+  func _performLegacyMatch(
```

Comment:
CaptureTests uses this kind of functionality, but there it's hand-invoking the VM. We could delete it, unless it's useful for a match-API-based test. I was considering taking the legacy stuff out of the `_StringProcessing` module and put it back into the `Prototypes` module, at which point I believe this would move with it.

I'll follow up after this PR.